lib/gh/client.ml

```diff
@@ -1,23 +1,26 @@
 module Json = Yojson.Basic.Util
 
+type error = No_github_token
+
 let github_api_url = "https://api.github.com/graphql"
 
 let query query_body =
-  let token = Sys.getenv "GITHUB_TOKEN" in
-
-  let response =
-    Ezcurl.post ~params:[]
-      ~headers:
-        [
-          ("Authorization", "bearer " ^ token);
-          ("User-Agent", "chshersh/github-tui");
-        ]
-      ~content:(`String (Printf.sprintf "{ \"query\": %S }" query_body))
-      ~url:github_api_url ()
-  in
-  match response with
-  | Error (_code, msg) -> Printf.sprintf "Error: %s" msg
-  | Ok response -> response.body
+  match Sys.getenv_opt "GITHUB_TOKEN" with
+  | None -> Error No_github_token
+  | Some token -> (
+      let response =
+        Ezcurl.post ~params:[]
+          ~headers:
+            [
+              ("Authorization", "bearer " ^ token);
+              ("User-Agent", "chshersh/github-tui");
+            ]
+          ~content:(`String (Printf.sprintf "{ \"query\": %S }" query_body))
+          ~url:github_api_url ()
+      in
+      match response with
+      | Error (_code, msg) -> Ok (Printf.sprintf "Error: %s" msg)
```

Comment:
Let's add one more constructor `Curl_error of { code: int; msg: string }` to `error` and return it here.

I don't think it should affect much of the rest code.